File: src/lib/k8s/provider.ts
export type CloudProvider = "eks" | "gke" | "aks" | "kubernetes";

export const PROVIDER_LABELS: Record<CloudProvider, string> = {
  eks: "Amazon EKS",
  gke: "Google GKE",
  aks: "Azure AKS",
  kubernetes: "Kubernetes",
};

export function detectCloudProvider(serverUrl: string, version?: string): CloudProvider {
  try {
    const hostname = new URL(serverUrl).hostname;
    if (hostname.endsWith(".eks.amazonaws.com")) return "eks";
    if (hostname.endsWith(".azmk8s.io")) return "aks";
    if (hostname.endsWith(".gke.goog") || hostname.includes("container.googleapis.com")) return "gke";
  } catch {
    // invalid URL — fall through
  }
  if (version) {
    if (version.includes("-gke.")) return "gke";
    if (version.includes("-eks-")) return "eks";
  }
  return "kubernetes";
}

File: backend/src/k8s/provider.rs
pub fn detect_cloud_provider(server_url: &str, version: Option<&str>) -> String {
    let url = server_url.to_lowercase();

    if url.contains(".eks.amazonaws.com") || url.contains(".elb.amazonaws.com") {
        return "eks".to_string();
    }
    if url.contains(".azmk8s.io") || url.contains("azure") {
        return "aks".to_string();
    }
    if url.contains(".gke.io") || url.contains("gke") {
        return "gke".to_string();
    }

    if let Some(v) = version {
        let v = v.to_lowercase();
        if v.contains("eks") {
            return "eks".to_string();
        }
        if v.contains("gke") {
            return "gke".to_string();
        }
        if v.contains("aks") {
            return "aks".to_string();
        }
    }

    "kubernetes".to_string()
}
